src/xdg.c

```diff
@@ -295,6 +295,16 @@ xdg_toplevel_view_close(struct view *view)
 	wlr_xdg_toplevel_send_close(xdg_toplevel_from_view(view));
 }
 
+static void
+xdg_toplevel_view_close_popups(struct view *view)
+{
+	struct wlr_xdg_toplevel *toplevel = xdg_toplevel_from_view(view);
+	struct wlr_xdg_popup *popup, *next;
+	wl_list_for_each_safe(popup, next, &toplevel->base->popups, link) {
+		wlr_xdg_popup_destroy(popup);
+	}
```

Comment:
I am not that familiar with our implementation for popups, does this also handle layered popups?
E.g. a menu spawning a submenu?

Reply:
Yes, indeed.